### ISSUE TITLE
Keyboard shortcuts

### ISSUE THREAD

**Issue by decibyte** [projecthamster/hamster#230]
From the overview window, it would be great to have some keyboard shortcuts for (suggestions in parentheses):
- New (Ctrl+N)
- Edit current (Ctrl+E)
- End current now (Ctrl+X)
- Focus search field (Ctrl+F)

In the edit window, Ctrl+Enter to save would be great  (although Alt+S works today), as this is sort of a de facto standard shortcut for clicking the OK button on things, at least on the web.

Support for optional global keyboard shortcuts for New, Edit, End would be even more useful.

**Comment by nchachereau** [comment]
Hi there!
Hamster is currently ongoing a rather large transformation. With the new interface, Ctrl+N and Ctrl+F both works in the Overview window, as does "Enter" (or Ctrl+Enter) in the Activity window. Escape will close the windows.

There is still room for more, which is why I leave this bug open. I'm rather unconvinced by Ctrl+X to stop tracking, as it usually means "cut" and someone might try to "cut" a focused activity.

Note that Hamster does not get much attention, as the project's creator does not maintain it any more, and no contributor manages to spend much time on its development.

**Comment by decibyte** [comment]
Ctrl+X was just a suggestion. The main point is that I'm missing some kbd shortcuts. Sounds good that some of it is already on the way. I'm looking forward to upcoming updates.

**Comment by ederag** [comment]
Thanks for the inputs.

#387 improve shortcuts and overall keyboard handling.
Feedback would be welcome.

What should be the best default shortcut for stop.
`Ctrl-S` sounds natural, but it is very often used to save in other software.
`Ctrl-H` (Halt) ?

**Comment by decibyte** [comment]
Ooh, lovely to see some attention here :)

It's a good question, as most of the obvious letters are taken. Maybe E (for "End")?

Just curious: Are you also working on global shortcuts?

**Comment by ederag** [comment]
> Just curious: Are you also working on global shortcuts?

Global shortcuts can be done using a combination of the desktop environment and
the command-line interface (e.g. `hamster overview` to popup the overview window).
So in the far future, at best.
Unless someone comes up  with a clean solution that works on any desktop
(not just Gnome).

> Maybe E (for "End")?

It is really tough to find a non-ambiguous one;  E could mean "Edit" as well.
What about `Ctrl+Space` ?
or `Ctrl+P` for "Pause" ?
Those two could be used also to "Play/unPause" the last activity ?
(much later; design issues foreseen)

**Comment by decibyte** [comment]
> Global shortcuts can be done using a combination of the desktop environment and
the command-line interface

Right. Don't waste your time on that.

> What about (...)

I kinda like `space`. Especially if it's used to toggle stop/start for the last activity, as it's used for exactly the same thing in a lot of timeline based software (music/video editing, etc.)

**Comment by ederag** [comment]
Those are good reasons to use `Space`, but alone it would be dangerous:
it happens sometimes to type while still being focused on the wrong window,
and notice it only after several words are typed in.
Prefixing with `Ctrl` mitigate that error. 

So thanks for the feedback, `Ctrl+Space` sounds good.
Of course, customizable controls would be better, 
but I do not know how to do that properly (in the correct Gtk3 way).

**Comment by decibyte** [comment]
Of course, you need the modifier key as well. It's just a fine mnemonic for anyone who knows `Space` from that kind of software.

Thank you for taking time to look at this :)

**Comment by ederag** [comment]
Done in #387.
Please open a new issue, linking to this one, as a follow-up,
if something is missing or wrong.
